ruff_lsp/server.py

```diff
@@ -750,6 +756,7 @@ def _get_severity(code: str) -> DiagnosticSeverity:
         "F821",  # undefined name `name`
         "E902",  # `IOError`
         "E999",  # `SyntaxError`
+        None,  # `SyntaxError` as of Ruff v0.5.0
```

Comment:
No, I meant that the `code` field in JSON would be either:
1. `E999` for all Ruff versions before v0.5
2. `None` for Ruff version v0.5 and later

Both are included in this list.

Reply:
> If you say it is required, does it mean that Ruff-LSP crashes or is it because it otherwise shows syntax errors as warnings?

Oh, I think I read your question wrong but yes otherwise it'll emit syntax errors as warnings. This is why I'm including `None` in this list so that it's emitted as error.

Reply:
> I guess there's not much we can do about it other than keeping to emit `E999` in the JSON emitter for syntax errors. It might be worth to do so for a while.

I'm not sure about that. If we do keep `E999` in the JSON emitter, it should probably be the same for other emitters as well otherwise users might find it confusing.

The current change will only break if there are any other diagnostics emitted by Ruff which doesn't have a rule code and is suppose to be a warning instead. I think it'll be a quite some time before this happens and by that time I think `ruff-lsp` will be deprecated.

Reply:
Makes sense. Let's ship as is